pkg/types/chain_writer.go

```diff
@@ -11,6 +11,9 @@ const (
 	ErrSettingTransactionGasLimitNotSupported = InvalidArgumentError("setting transaction gas limit is not supported")
 )
 
+type ContractWriter = ChainWriter
+
+// Deprecated: Use ContractWriter instead.
 type ChainWriter interface {
```

Comment:
Swapping these makes the "correct" type more direct and sets up removal to be a little cleaner later :shrug: 
```suggestion
// Deprecated: Use ContractWriter instead.
type ChainWriter = ContractWriter

type ContractWriter interface {
```

Reply:
ah, didn't realise that was legal